src/GeoCoding/OrbitState.jl

```diff
@@ -5,3 +5,102 @@ struct OrbitState
     velocity::Array{Float64,1}
 end 
 
+
+get_speed(state::OrbitState) = LinearAlgebra.norm(state.velocity)
+
+
+get_burst_mid_states(image::SingleLookComplex, interpolator) = [interpolator(t) 
+                                                            for t in get_burst_mid_times(image)]
+
+
+"""
+orbit_state_interpolator(orbit_states::Vector{OrbitState}, image::SARImage, 
+    polynomial_degree::Integer=4, margin::Integer = 3 )
+
+    Create a polynomial interpolation functions for orbit states valid in the time span
+    from image start time to image end time.
+
+    #Returns
+    Anonymous interpolation function. (Input: DateTime, Output: OrbitState)
+"""
+function orbit_state_interpolator(orbit_states::Vector{OrbitState}, image::SARImage, 
+    polynomial_degree::Integer=4, margin::Integer = 3 )
+    time_range = get_time_range(image)
+    
+    # check that the orbit states cover the image
+    @assert (orbit_states[1].time < time_range[1]) &&  (time_range[2] < orbit_states[end].time)
+    
+    orbit_state_interpolator(orbit_states, time_range; 
+    polynomial_degree=polynomial_degree, margin = margin )
+end
+
+function orbit_state_interpolator(orbit_states::Vector{OrbitState}, time_range::Tuple{DateTime,DateTime}; 
+    polynomial_degree::Integer=4, margin::Integer = 3 )
+
+
+    #select orbit states
+    selected_orbit_states = _select_orbit_states(orbit_states, time_range, margin)
+    @assert length(selected_orbit_states) > polynomial_degree
```

Comment:
similar, I would add
`@assert false "Expected polynomial degree to be smaller than length of selected orbit states"`

Reply:
This is maybe more obvious from the assert itself than above. But still I guess it is nice with a small message :)

Reply:
I do not think you can add a message to @assert. I can do an if statement and throw an error

Reply:
I looked it up before I wrote it, I am pretty sure you can, but never mind :)
https://docs.julialang.org/en/v1/base/base/#Base.@assert

Reply:
You are right :) 
I learn something new all the time